Cubical/Categories/Displayed/Section/Base.agda

```diff
@@ -0,0 +1,307 @@
+
+-}
+{-# OPTIONS --safe #-}
+module Cubical.Categories.Displayed.Section.Base where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Data.Sigma
+import      Cubical.Data.Equality as Eq
+
+open import Cubical.Categories.Category
+open import Cubical.Categories.Functor.Base
+open import Cubical.Categories.Functor.Equality
+open import Cubical.Categories.Displayed.Base
+open import Cubical.Categories.Displayed.Functor
+import      Cubical.Categories.Displayed.Reasoning as HomᴰReasoning
+
+private
+  variable
+    ℓB ℓB' ℓBᴰ ℓBᴰ' ℓC ℓC' ℓCᴰ ℓCᴰ' ℓD ℓD' ℓDᴰ ℓDᴰ' ℓE ℓE' ℓEᴰ ℓEᴰ' : Level
+
+open Category
+open Categoryᴰ
+open Functor
+
+module _ {C : Category ℓC ℓC'}
+         {D : Category ℓD ℓD'}
+         (F : Functor D C)
+         (Cᴰ : Categoryᴰ C ℓCᴰ ℓCᴰ')
+         where
+  private
+    module C = Category C
+    module D = Category D
+    module Cᴰ = Categoryᴰ Cᴰ
+    module F = Functor F
+
+  -- Section without a qualifier means *local* section.
+  record Section : Type (ℓ-max (ℓ-max ℓC ℓC')
+                        (ℓ-max (ℓ-max ℓD ℓD')
+                        (ℓ-max ℓCᴰ ℓCᴰ')))
+    where
+    field
+      F-obᴰ  : ∀ d → Cᴰ.ob[ F ⟅ d ⟆ ]
+      F-homᴰ : ∀ {d d'} (f : D.Hom[ d , d' ])
+        → Cᴰ.Hom[ F ⟪ f ⟫ ][ F-obᴰ d , F-obᴰ d' ]
+      F-idᴰ : ∀ {d} → F-homᴰ (D.id {d}) Cᴰ.≡[ F .F-id ] Cᴰ.idᴰ
+      F-seqᴰ : ∀ {d d' d''}
+            → (f : D.Hom[ d , d' ])(g : D.Hom[ d' , d'' ])
+            → F-homᴰ (f D.⋆ g) Cᴰ.≡[ F .F-seq f g ] F-homᴰ f Cᴰ.⋆ᴰ F-homᴰ g
+
+{-
+   A Global Section is a local section over the identity functor.
+
+          Cᴰ
+        ↗ |
+       /  |
+    s /   |
+     /    |
+    /     ↓
+   C ==== C
+
+
+   So global sections are by definition local sections
+
+   All local sections can be implemented as global sections into the
+   reindexed displayed category. See Reindex.agda for details.
+
+   But this isomorphism is not definitional (i.e. the equations are
+   not provable by refl). Constructing a local section is preferable
+   as they are more flexible, but often elimination principles are
+   naturally proven first about global sections and then generalized
+   to local sections using reindexing.
+
+   It is good style is to offer both: elimLocal to construct a local
+   section and elimGlobal to construct a global section.
+-}
+
+module _ {C : Category ℓC ℓC'}
+         (Cᴰ : Categoryᴰ C ℓCᴰ ℓCᴰ')
+         where
+  GlobalSection : Type _
+  GlobalSection = Section Id Cᴰ
+
+-- Reindexing a section.
+module _ {C : Category ℓC ℓC'}
+         {D : Category ℓD ℓD'}
+         {Cᴰ : Categoryᴰ C ℓCᴰ ℓCᴰ'}
+         where
+  open Section
+  private
+    module Cᴰ = Categoryᴰ Cᴰ
+    module R = HomᴰReasoning Cᴰ
+  reindS : ∀ {F G}(H : Path (Functor D C) F G) → Section F Cᴰ → Section G Cᴰ
+  reindS H = subst (λ F → Section F Cᴰ) H
+
+  reindS' : ∀ {F G : Functor D C} (H : FunctorEq F G)
```

Comment:
The difference is in the definitional behavior: `reindS` is defined by `subst` and so is guaranteed to transport when used. OTOH if you use `reindS'` and your `FunctorEq` is given by `refl , refl` (as it is in every use in this PR) then there will be no transport. I expect all uses of `reindS'` to be of this form, otherwise I would recommend `reindS`. I will add a comment explaining the situation.